### PR TITLE
smlnj: 110.91 -> 110.95, add 64-bit support

### PR DESCRIPTION
**Pull request by aske**
###### Motivation for this change
Update package, use 64-bit support added in 110.94.
Also drop isDarwin check in `top-level/all-packages.nix` till darwin build is fixed.

I needed to patch `heap2exec` for it to run after installation; I filed an upstream bug [here](http://smlnj-gforge.cs.uchicago.edu/tracker/index.php?func=detail&aid=237&group_id=33&atid=215).

With these changes on `x86_64-linux` only 64-bit version is installed , though `smlnj` also supports simultaneous 32 and 64 bit installation. Would it be useful to add support for joint installation?

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @thoughtpolice

### CODE REVIEW

**Comment by aske** [comment]
Hm, this seems to break even the latest revision of `manticore`, I'll investigate further a bit later.

**Comment by aske** [comment]
Fixed `manticore` with new smlnj, though let's see if these fixes are right (https://github.com/ManticoreProject/manticore/issues/17).

**Comment by aske** [comment]
Alright, with all changes accepted by upstream this PR is now ready.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -8723,9 +8723,7 @@ in
   shmig = callPackage ../development/tools/database/shmig { };
 
   smlnjBootstrap = callPackage ../development/compilers/smlnj/bootstrap.nix { };
-  smlnj = if stdenv.isDarwin
-            then callPackage ../development/compilers/smlnj { }
-            else pkgsi686Linux.callPackage ../development/compilers/smlnj { };
```
Comment (thoughtpolice) [inline pkgs/top-level/all-packages.nix]: 💯

**Review by thoughtpolice — APPROVED** [review]
LGTM

**Comment by thoughtpolice** [comment]
@GrahamcOfBorg build smlnj

**Pull request merged**